gossip/src/crds.rs

```diff
@@ -147,14 +147,20 @@ impl Cursor {
 }
 
 impl VersionedCrdsValue {
-    fn new(value: CrdsValue, cursor: Cursor, local_timestamp: u64) -> Self {
+    fn new(value: CrdsValue, cursor: Cursor, local_timestamp: u64, route: GossipRoute) -> Self {
         let value_hash = hash(&serialize(&value).unwrap());
+        let num_push_dups = match route {
+            // set num_push_dups to 2^8 to indicate PullResponse
+            // unlikely a node receives the exact same message via push 2^8 times
+            GossipRoute::PullResponse => u8::MAX,
```

Comment:
These kind of sentinel values are pretty risky and can introduce subtle bugs if someone later introduces a change unaware of this u8::MAX special casing.
I am more inclined if we just change `num_push_dups` to `num_push_recv` (or something like that), and simply count how many times this value was received through the push path.

Reply:
so when `ReceivedCache` uses `num_push_recv ` instead of `num_push_dups`, should we then look at `num_push_recv - 1` in `ReceivedCacheEntry::record()`: https://github.com/anza-xyz/agave/blob/3863bb1bdf0f7c9a0b35c2c19dc50943ca39657e/gossip/src/received_cache.rs#L81?